src/tron/tronEngine.js

```diff
@@ -0,0 +1,548 @@
+      this.currencyEngineCallbacks.onBalanceChanged(tk, balance)
+    }
+    this.tokenCheckBalanceStatus[tk] = 1
+    this.updateOnAddressesChecked()
+  }
+
+  async checkAccountInnerLoop () {
+    const address = this.walletLocalData.publicKey
+    const url = '/v1/accounts/'
+    const finalUrl = `${url}${tronWeb.address.toHex(address)}`
+    try {
+      const jsonObj = await this.multicastServers('trx_getBalance', finalUrl)
+      const valid = validateObject(jsonObj, TronApiAccountBalance)
+      // this.log('checkAccountInnerLoop acc balance jsonObj: ', jsonObj)
+      // if (valid) {
+      //   this.updateBalance('TRX', jsonObj.data[0].balance.toString())
+      //   for (const tk of this.walletLocalData.enabledTokens) {
+      //     for (const balance of jsonObj.data[0].asset) {
+      //       if (balance.key === tk) {
+      //         const denom = getDenomInfo(this.currencyInfo, tk)
+      //         if (!denom) {
+      //           this.log(`Received unsupported currencyCode: ${tk}`)
+      //           break
+      //         }
+      //         const nativeAmount = bns.mul(balance.value, denom.multiplier)
+      //         this.updateBalance(tk, nativeAmount)
+      //       }
+      //     }
+      //   }
+      // }
+      if (valid) {
+        this.updateBalance('TRX', jsonObj.data[0].balance.toString())
+      } else {
+        this.log(
+          'checkAccountInnerLoop: Error getting balance. Received: ',
+          jsonObj.data
+        )
+      }
+    } catch (e) {
+      this.log('Error checking TRX address balance: ', e)
+    }
+  }
+
+  processTronApiTransaction (tx: TronApiTransaction, currencyCode: string) {
+    const publicKey: string = this.walletLocalData.publicKey.toLowerCase()
+    let netNativeAmount: string // Amount received into wallet
+    const ourReceiveAddresses: Array<string> = []
+    const nativeNetworkFee: string = '0'
+    const nativeValue: string = tx.amount.toString()
+    const fromAddr: string = tronWeb.address
+      .fromHex(tx.owner_address)
+      .toLowerCase()
+    const toAddr: string = tronWeb.address.fromHex(tx.to_address).toLowerCase()
+
+    // isSpend Tx
+    if (fromAddr === publicKey) {
+      // if it's a send to one's self
+      if (tx.owner_address === toAddr) {
+        // Spend to self. netNativeAmount is just the fee
+        netNativeAmount = bns.mul(nativeNetworkFee, '-1')
+      } else {
+        // set amount to spending amount
+        netNativeAmount = bns.mul(nativeValue, '-1')
+      }
+    } else {
+      // Receive transaction
+      netNativeAmount = bns.add('0', nativeValue)
+      // ourReceiveAddresses.push(this.walletLocalData.publicKey.toLowerCase())
+    }
+
+    const otherParams: Object = {}
+
+    let blockHeight = tx.blockNumber
+    if (blockHeight < 0) blockHeight = 0
+    const unixTimestamp = new Date(tx.block_timestamp)
+    const edgeTransaction: EdgeTransaction = {
+      txid: tx.txID,
+      date: unixTimestamp.getTime(),
```

Comment:
This is setting a JavaScript timestamp which is incorrect as the Edge transaction object is expecting a UNIX timestamp. You need to divide by 1000 to get a UNIX timestamp.I think the problem is being hidden by the fact that you are passing a UNIX timestamp to the new date constructor which is expecting a JavaScript timestamp. I think this bug also exists in BNB so I see where you carried over the issue. If the timestamp from the API is already in a UNIX number format, then you can simply assign that directly to the date with no additional conversion